libvast/src/system/partition.cpp

```diff
@@ -55,6 +55,10 @@ partition::partition(index_state* state, uuid id, size_t max_capacity)
 }
 
 partition::~partition() noexcept {
+  for (auto kv : indexers_) {
+    auto actor = kv.second.indexer;
+    state_->self->send(actor, stop_atom::value);
```

Comment:
I was wondering if we could just send `shutdown_atom` here and get rid of the extra `stop_atom`, but I wasnt completely sure if it is guaranteed that no lookups are still in progress when the partition is destroyed here.

Reply:
We might not need to fiddle with the shutdown logic at all if we reduce the scope of the telemetry to the streaming sink. Would it be possible to start the telemetry heartbeat only when `make_sink` is called and then shut it down in the stop handler?

Reply:
I think that's what I originally tried and it caused some trouble, but I'll have to try again because I can't remember what exactly the problem was (and if it was maybe caused by something else)

Reply:
It worked now, so it looks like the problem was something else.